uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Page/WorkPattern.php

```diff
@@ -170,4 +177,16 @@ private function calculateLinksMask($workPattern) {
 
     return $mask;
   }
+
+  /**
+   * Checks whether a WorkPattern object cannot be deleted.
+   *
+   * @param int $workPatternID
+   *
+   * @return bool
+   */
+  private function canNotDelete($workPatternID) {
+    $workPattern = new WorkPatternService();
```

Comment:
Given that this will be called for each record in the list, it's better to instantiate the service once, probably in the constructor or inside the `run` method